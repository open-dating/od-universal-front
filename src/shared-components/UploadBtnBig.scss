.upload-btn-big{
  margin: 0 auto;
  padding: 5px;
  border: 2px dashed #3f51b5;
  border-radius: 5px;
  height: 200px;
  width: 200px;

  .upload-btn-big__btn{
    position: relative;
    height: 100%;
    width: 100%;

    svg {
      height: 50px;
      width: 50px;
    }
  }

  input[type="file"] {
    position: absolute;
    top: 0;
    left: 50%;
    transform: translate(-50%, 0);
    opacity: 0;
    height: 210px;
    width: 210px;
  }
}
