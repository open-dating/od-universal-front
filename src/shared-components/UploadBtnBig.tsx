import React, {ChangeEvent} from 'react'
import Button from '@material-ui/core/Button'
import AddPhotoAlternateIcon from '@material-ui/icons/AddPhotoAlternate'
import {getRandom} from './../utils/random'

import './UploadBtnBig.scss'

export function UploadBtnBig({loading, onChange}: {loading?: boolean, onChange: (evt: ChangeEvent<HTMLInputElement>) => void}) {
  // always render new button, for upload duplicate photos
  const uniqKey = getRandom()

  return (
    <div className="upload-btn-big">
      <Button
        variant="contained"
        color="primary"
        disabled={loading}
        className="upload-btn-big__btn"
      >
        <AddPhotoAlternateIcon/>
      </Button>
      <input
        key={uniqKey}
        type="file"
        onChange={onChange}
      />
    </div>
  )
}
