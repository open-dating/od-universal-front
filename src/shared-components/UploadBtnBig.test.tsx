import React, {ChangeEvent} from 'react'
import {shallow} from 'enzyme'
import {UploadBtnBig} from './UploadBtnBig'

jest.mock('./../utils/random', () => ({
    getRandom: () => 1,
}))

const props = { onChange: (evt: ChangeEvent<HTMLInputElement>) => {} }
let loading = false

const wrapper = shallow(<UploadBtnBig {...props} loading={loading}/>)

describe('<UploadBtnBig /> rendering', ()=> {
    it('should render 1 div', () => {
        expect(wrapper.find('div')).toHaveLength(1)
    })
})

describe('<UploadBtnBig /> interactions', () => {
    it('should render 1 button with false', () => {
        wrapper.setProps({loading: loading})
        expect(wrapper).toMatchSnapshot()
    })

    it('should render 1 button with true', () => {
        wrapper.setProps({loading: !loading})
        expect(wrapper).toMatchSnapshot()
    })
})


   