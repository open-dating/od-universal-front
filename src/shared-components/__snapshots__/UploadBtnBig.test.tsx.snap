// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UploadBtnBig /> interactions should render 1 button with false 1`] = `
<div
  className="upload-btn-big"
>
  <WithStyles(ForwardRef(Button))
    className="upload-btn-big__btn"
    color="primary"
    disabled={false}
    variant="contained"
  >
    <AddPhotoAlternateIcon />
  </WithStyles(ForwardRef(Button))>
  <input
    key="1"
    onChange={[Function]}
    type="file"
  />
</div>
`;

exports[`<UploadBtnBig /> interactions should render 1 button with true 1`] = `
<div
  className="upload-btn-big"
>
  <WithStyles(ForwardRef(Button))
    className="upload-btn-big__btn"
    color="primary"
    disabled={true}
    variant="contained"
  >
    <AddPhotoAlternateIcon />
  </WithStyles(ForwardRef(Button))>
  <input
    key="1"
    onChange={[Function]}
    type="file"
  />
</div>
`;
